### ISSUE TITLE
error: ‘afp_spotlight_rpc’ undeclared

### ISSUE THREAD

**Issue by i0nel** [Netatalk/netatalk#654]
Hi,

Getting the following error message:

`auth.c: In function ‘set_auth_switch’:
auth.c:193:81: error: ‘afp_spotlight_rpc’ undeclared (first use in this function)
  193 |             uam_afpserver_action(AFP_SPOTLIGHT_PRIVATE, UAM_AFPSERVER_POSTAUTH, **afp_spotlight_rpc**, NULL);
      |                                                                                 ^~~~~~~~~~~~~~~~~
auth.c:193:81: note: each undeclared identifier is reported only once for each function it appears in
make[3]: *** [Makefile:745: afpd-auth.o] Error 1

make[3]: Leaving directory '/home/x/netatalk-code/etc/afpd'
make[2]: *** [Makefile:1221: install] Error 2
make[2]: Leaving directory '/home/x/netatalk-code/etc/afpd'
make[1]: *** [Makefile:466: install-recursive] Error 1
make[1]: Leaving directory '/home/x/netatalk-code/etc'
make: *** [Makefile:522: install-recursive] Error 1`

Any help is appreciated

**Comment by rdmark** [comment]
@i0nel What version of the netatalk code are you trying to build? What operating system are you building on? Also, it would be helpful to know the parameters that you passed to the `configure` script before compiling, as well as the summary output from the configure script.

This particular error is because `include/atalk/spotlight.h` wasn't included by the compiler, which is most likely because the macro that checks for Spotlight prerequisites malfunctioned. We recently made some improvements in that very area in the `main` branch, so I'm curious to know if you're using slightly outdated code, or if there's a corner case where the Spotlight detection still malfunctions.

**Comment by i0nel** [comment]
@rdmark 
OS: **Debian 12** (bookworm)


Netatalk-code Verion: **3.1.99dev**


user@host:~/netatalk-code$ **./configure --with-init-style=debian-systemd  --with-tracker-pkgconfig-version=3.0**


```
Compilation summary:
    CPPFLAGS       = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_srcdir) 
    CFLAGS         = -D_U_="__attribute__((unused))" -g -O2
    LIBS           = 
    PTHREADS:
        LIBS   = -lpthread
        CFLAGS = -pthread
    TRACKER:
        LIBS   = -ltracker-sparql-3.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 
        CFLAGS = -I/usr/include/tracker-3.0 -I/usr/include/tracker-3.0/libtracker-sparql -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/json-glib-1.0 -pthread -I/usr/include/libxml2 
    SSL:
        LIBS   =  -L/usr/lib64 -lcrypto
        CFLAGS =  -I/usr/include/openssl
    LIBGCRYPT:
        LIBS   = -L/usr/lib/x86_64-linux-gnu -lgcrypt
        CFLAGS = 
    BDB:
        LIBS   =  -L/usr/lib64 -ldb-5.3
        CFLAGS = 
    ZEROCONF:
        LIBS   =  -lavahi-common -lavahi-client
        CFLAGS =  -D_REENTRANT
    LIBEVENT:
        LIBS   = -levent 
        CFLAGS = 
    TALLOC:
        LIBS   = 
        CFLAGS = 
Configure summary:
    INIT STYLE:
         debian-systemd
    AFP:
         Extended Attributes: ad | sys
         ACL support: no
         Spotlight: no
    CNID:
         backends:  dbd last
    UAMS:
         DHX     ( SHADOW)
         DHX2    ( SHADOW)
         RANDNUM (afppasswd)
         clrtxt  ( SHADOW)
         guest
    Options:
         Zeroconf support:        yes
         tcp wrapper support:     no
         valid shell check:       yes
         cracklib support:        no
         ACL support:             auto
         Kerberos support:        auto
         LDAP support:            no
         AFP stats via dbus:      yes
         dtrace probes:           no
    Paths:
         Netatalk lockfile:       /var/lock/netatalk
         init directory:          /lib/systemd/system
         dbus system directory:   /usr/local/etc/dbus-1/system.d
    Documentation:
         Docbook:                 no

```

**Comment by rdmark** [comment]
@dgsga Do we need to change more (all?) preprocessor macros from HAVE_TRACKER to WITH_SPOTLIGHT? It seems this user’s environment has the former evaluate to the and the latter to false.

f.e. https://github.com/Netatalk/netatalk/blob/a478adc4c3bd6c63f5d65b5ba589361e41557d7e/etc/afpd/auth.c#L192

**Comment by dgsga** [comment]
@rdmark, I think you're right. The HAVE_TRACKER conditional has been superseded by WITH_SPOTLIGHT. We only need the HAVE_TRACKER3 conditional to determine whether the tracker version installed is 3 vs. 1 or 2. See PR #661

**Comment by dgsga** [comment]
> @rdmark OS: **Debian 12** (bookworm)
> 
> Netatalk-code Verion: **3.1.99dev**
> 
> user@host:~/netatalk-code$ **./configure --with-init-style=debian-systemd --with-tracker-pkgconfig-version=3.0**
> 
> ```
> Compilation summary:
>     CPPFLAGS       = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_srcdir) 
>     CFLAGS         = -D_U_="__attribute__((unused))" -g -O2
>     LIBS           = 
>     PTHREADS:
>         LIBS   = -lpthread
>         CFLAGS = -pthread
>     TRACKER:
>         LIBS   = -ltracker-sparql-3.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 
>         CFLAGS = -I/usr/include/tracker-3.0 -I/usr/include/tracker-3.0/libtracker-sparql -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/json-glib-1.0 -pthread -I/usr/include/libxml2 
>     SSL:
>         LIBS   =  -L/usr/lib64 -lcrypto
>         CFLAGS =  -I/usr/include/openssl
>     LIBGCRYPT:
>         LIBS   = -L/usr/lib/x86_64-linux-gnu -lgcrypt
>         CFLAGS = 
>     BDB:
>         LIBS   =  -L/usr/lib64 -ldb-5.3
>         CFLAGS = 
>     ZEROCONF:
>         LIBS   =  -lavahi-common -lavahi-client
>         CFLAGS =  -D_REENTRANT
>     LIBEVENT:
>         LIBS   = -levent 
>         CFLAGS = 
>     TALLOC:
>         LIBS   = 
>         CFLAGS = 
> Configure summary:
>     INIT STYLE:
>          debian-systemd
>     AFP:
>          Extended Attributes: ad | sys
>          ACL support: no
>          Spotlight: no
>     CNID:
>          backends:  dbd last
>     UAMS:
>          DHX     ( SHADOW)
>          DHX2    ( SHADOW)
>          RANDNUM (afppasswd)
>          clrtxt  ( SHADOW)
>          guest
>     Options:
>          Zeroconf support:        yes
>          tcp wrapper support:     no
>          valid shell check:       yes
>          cracklib support:        no
>          ACL support:             auto
>          Kerberos support:        auto
>          LDAP support:            no
>          AFP stats via dbus:      yes
>          dtrace probes:           no
>     Paths:
>          Netatalk lockfile:       /var/lock/netatalk
>          init directory:          /lib/systemd/system
>          dbus system directory:   /usr/local/etc/dbus-1/system.d
>     Documentation:
>          Docbook:                 no
> ```

@i0nel, this PR should fix your issue. You need to have tracker, libtalloc-dev and libtracker-sparql-3.0-dev installed on Bookworm for Spotlight support to be compiled successfully.

**Comment by dgsga** [comment]
> @rdmark OS: **Debian 12** (bookworm)
> 
> Netatalk-code Verion: **3.1.99dev**
> 
> user@host:~/netatalk-code$ **./configure --with-init-style=debian-systemd --with-tracker-pkgconfig-version=3.0**
> 
> ```
> Compilation summary:
>     CPPFLAGS       = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_srcdir) 
>     CFLAGS         = -D_U_="__attribute__((unused))" -g -O2
>     LIBS           = 
>     PTHREADS:
>         LIBS   = -lpthread
>         CFLAGS = -pthread
>     TRACKER:
>         LIBS   = -ltracker-sparql-3.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 
>         CFLAGS = -I/usr/include/tracker-3.0 -I/usr/include/tracker-3.0/libtracker-sparql -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/json-glib-1.0 -pthread -I/usr/include/libxml2 
>     SSL:
>         LIBS   =  -L/usr/lib64 -lcrypto
>         CFLAGS =  -I/usr/include/openssl
>     LIBGCRYPT:
>         LIBS   = -L/usr/lib/x86_64-linux-gnu -lgcrypt
>         CFLAGS = 
>     BDB:
>         LIBS   =  -L/usr/lib64 -ldb-5.3
>         CFLAGS = 
>     ZEROCONF:
>         LIBS   =  -lavahi-common -lavahi-client
>         CFLAGS =  -D_REENTRANT
>     LIBEVENT:
>         LIBS   = -levent 
>         CFLAGS = 
>     TALLOC:
>         LIBS   = 
>         CFLAGS = 
> Configure summary:
>     INIT STYLE:
>          debian-systemd
>     AFP:
>          Extended Attributes: ad | sys
>          ACL support: no
>          Spotlight: no
>     CNID:
>          backends:  dbd last
>     UAMS:
>          DHX     ( SHADOW)
>          DHX2    ( SHADOW)
>          RANDNUM (afppasswd)
>          clrtxt  ( SHADOW)
>          guest
>     Options:
>          Zeroconf support:        yes
>          tcp wrapper support:     no
>          valid shell check:       yes
>          cracklib support:        no
>          ACL support:             auto
>          Kerberos support:        auto
>          LDAP support:            no
>          AFP stats via dbus:      yes
>          dtrace probes:           no
>     Paths:
>          Netatalk lockfile:       /var/lock/netatalk
>          init directory:          /lib/systemd/system
>          dbus system directory:   /usr/local/etc/dbus-1/system.d
>     Documentation:
>          Docbook:                 no
> ```

@i0nel, PR #664 should fix the issue. Please note that **libtalloc-dev**, **libtracker-sparql-3.0-dev** and **tracker** must be installed for Spotlight support.

**Comment by i0nel** [comment]
Deleted all older files

Pulled latest (revised) version. 

then configured the build only with this parameter:
user@host:~/netatalk-code$ ./configure --with-init-style=debian-systemd


Seems to be installed NOW:
```
user@host:~$	sudo afpd -V

afpd 3.1.99dev - Apple Filing Protocol (AFP) daemon of Netatalk

afpd has been compiled with support for these features:

          AFP versions:	2.2 3.0 3.1 3.2 3.3 3.4 
         CNID backends:	dbd last 
      Zeroconf support:	Avahi
  TCP wrappers support:	No
   Admin group support:	Yes
    Valid shell checks:	Yes
      cracklib support:	No
            EA support:	ad | sys
           ACL support:	No
          LDAP support:	No
         D-Bus support:	Yes
     Spotlight support:	No
         DTrace probes:	No

              afp.conf:	/usr/local/etc/afp.conf
           extmap.conf:	/usr/local/etc/extmap.conf
       state directory:	/usr/local/var/netatalk/
    afp_signature.conf:	/usr/local/var/netatalk/afp_signature.conf
      afp_voluuid.conf:	/usr/local/var/netatalk/afp_voluuid.conf
       UAM search path:	/usr/local/lib/netatalk/
  Server messages path:	/usr/local/var/netatalk/msg/
```


Thank you ALL for your time
